### DISCUSSION TITLE
Initialize a cache once during server init?

### DISCUSSION THREAD

**Discussion by jaidetree** [remix-run/remix#2949] [Q&A] [answered]
Got a `posts.server.ts` that has an API for fetching and caching posts from a source.

Would like to have this run asynchronously once when the remix server starts so content is pre-cached. Originally, tried to create a standalone script to run during build but this proved impractical as a lot of code depends on the aliases that the tsconfig specifies that are not supported by a regular node js script file.

Where would it live? Was originally thinking it could be placed in `posts.server.ts` or `entry.server.ts` but both those seem to run every request. 

Is there a good place to do something like that?

**Comment by kiliman** [comment, marked as answer]
The reason module-level variables are reinitialized on every request is due to `<LiveReload/>`

In development you should store your cache on the `global` object to survive the purge.

See the Jokes tutorial where it discusses how to maintain the database connection.

**Reply by jaidetree** [reply]
Thanks! That's where I landed on using a `global.cacheInitialized`:

```ts
if (!global.cacheInitialized) {
  // cache content

  global.cacheInitialized = true
}
```
